Source/WebCore/page/EventHandler.cpp

```diff
@@ -4320,19 +4325,19 @@ bool EventHandler::handleDrag(const MouseEventWithHitTestResults& event, CheckDr
     auto hasNonDefaultPasteboardData = HasNonDefaultPasteboardData::No;
     
     if (dragState().shouldDispatchEvents) {
-        ASSERT(dragState().source);
-        auto dragStartDataTransfer = DataTransfer::createForDragStartEvent(dragState().source->document());
+        ASSERT(draggedElement());
+        auto dragStartDataTransfer = DataTransfer::createForDragStartEvent(draggedElement()->protectedDocument());
         m_mouseDownMayStartDrag = dispatchDragStartEventOnSourceElement(dragStartDataTransfer);
         if (downcast<StaticPasteboard>(dragStartDataTransfer->pasteboard()).hasNonDefaultData())
             hasNonDefaultPasteboardData = HasNonDefaultPasteboardData::Yes;
         dragState().dataTransfer->moveDragState(WTFMove(dragStartDataTransfer));
 
-        if (dragState().source && dragState().type == DragSourceAction::DHTML && !dragState().dataTransfer->hasDragImage()) {
-            dragState().source->document().updateStyleIfNeeded();
-            if (auto* renderer = dragState().source->renderer()) {
+        if (draggedElement() && dragState().type == DragSourceAction::DHTML && !dragState().dataTransfer->hasDragImage()) {
```

Comment:
Or at the very least, in a block like this where we check for `draggedElement()` then do things with it, why not get a `protectedDragElement()` here and hold it while inside this block (and perhaps `WTFMove()` into `setDragImage`?)

Reply:
I'll look. We may indeed benefit by storing the dragElement in a variable .

Reply:
I did the WTFMove()